### DISCUSSION TITLE
OpenCV.Native.Net Error when I try to Run Pylon Viewer Camera

### DISCUSSION THREAD

**Discussion by ab4728** [bonsai-rx/bonsai#1574] [Q&A]
Is anyone familiar with this error? Both the camera manufacturers and the company I'm using Bonsai for have said this is "out of their scope." 

![image](https://github.com/bonsai-rx/bonsai/assets/111145910/11b0d366-0f20-4133-a5eb-3c3f09b58c8a)

**Comment by bruno-f-cruz** [comment]
Can you try to re-install bonsai from scratch? (i.e. uninstall, delete the folder, download a new release from bonsai-rx). If that does not fix it, [download `Visual C++ Redistributable for Visual Studio 2012`](https://www.microsoft.com/en-us/download/details.aspx?id=30679)(https://www.microsoft.com/en-us/download/details.aspx?id=30679)

**Reply by ab4728** [reply]
Hi, I've tried running the redistributable but I get this error

![image](https://github.com/bonsai-rx/bonsai/assets/111145910/7ed2e555-b41b-48b3-adb4-ad3136451901)

**Reply by bruno-f-cruz** [reply]
Which distribution did you download and what is the architecture of your PC (arm, 64bit, 32bit)?

**Reply by ab4728** [reply]
I'm not sure. I think it was "arm" that was causing that error because I just successfully installed x64 and x86. Unfortunately, neither of those solved the issue. Attached is a screenshot of command window when the error is thrown. Notably, the path referenced is not one we have on this computer. 

![image](https://github.com/bonsai-rx/bonsai/assets/111145910/712ff5e2-9c36-4de9-a7a6-b338d64ccb6e)

**Reply by bruno-f-cruz** [reply]
I am afraid i cant help you with that error as I am not familiar with the specific package. I would urge you to get in touch with the authors.

**Reply by ab4728** [reply]
Okay, who are the authors in this case? It seems like there are errors from multiple sources.

**Reply by bruno-f-cruz** [reply]
All the errors shown in the previous window seem to come from the same package "Neurophotometrics". Their repository seems to be here:
https://github.com/neurophotometrics/neurophotometrics

**Reply by ab4728** [reply]
I see. Interesting. They directed me here..

**Reply by bruno-f-cruz** [reply]
Uhm sorry for the understandable frustration, but we do not maintain that package. Hopefully, someone else who has had this problem or has the time to debug it with you might be able to help!

**Reply by ab4728** [reply]
I understand. Thank you for your help I really appreciate it.

**Reply by SaiK238** [reply]
Did you find a solution to this? We ran into the same issue and followed the suggestions by @bruno-f-cruz but still getting the same error.

**Comment by ab4728** [comment]
Hey,

I'm sorry to hear you're having this issue. Unfortunately, we did not find a fix for this. Our alternative was to buy another camera with a USB connection.


Alexander Benson

Drexel University College of Medicine

Department of Neurobiology and Anatomy, Gao Laboratory

Room # 255 Queen Lane Campus

he/him/his

________________________________

**Comment by romainligneul** [comment]
Based on the screenshot (and the reproduction of this error on my own system), it seems that you've forgotten to provide a parameter file. 
You can export it from Pylon viewer in Camera->SaveFeatures, to load it afterwards in Bonsai.

**Comment by seanpaulbradley** [comment]
We have the same issue using either a Basler Ace or Ace2Pro.  Both are GigE and functional using scripts or PylonViewer.

The layout is simply a Python node (default .pfs + SN provided), the image source, and either a videowriter or publishsubject node.  Both return the same error mentioned here.

I've ensured I'm running Pylon 7.1, re-installed .NET 2012, deleted/re-downloaded Bonsai.  No dice.

The wrinkle on our end is that I'm only able to use the portable version - the installer doesn't work.  I'm 99% sure that's an IT issue on our end.

I'm not sure how to read the error, so I'm not sure what else to try for troubleshooting.

**Comment by ab4728** [comment]
I'm glad to see people engaging with this, I hope the issue gets resolved soon. Our lab's fix was to get a camera with a USB connection, as GigE did not work.

[cid:c6557dd3-4f0b-4e83-a55e-4c00cc82d92a]
________________________________

**Reply by Andrianarivelo** [reply]
Hi, we have exactly the same problem with a acA2040-90umNIR USB camera. How did you manage to make it work with an usb camera ? Is there a fix ? Can the maker of the pylon capture help us ?
